### ISSUE TITLE
Height function

### ISSUE THREAD

**Issue by simplesmiler** [Akryum/vue-virtual-scroller#63]
**TL;DR:** Ability to provide a "height function" instead of setting height on an item.

The API might be the following: `<recycle-list :items="items" :height-fn="item => heights[item.id]">`.

---

My use case is the following. Items come from a computed property `items`, and they have variable height with a known approximation (e.g. about 50±20px). The exact height is not known until the component is rendered.

Because of variable height, components emit their desired height for the given item once it is known.

But because items come from a computed property, I can not directly set the height on the item object. Instead I have a data field `heights`, and the computed property `items` populates heights from this data field.

The downside is that `items` list is recalculated every time `heights` is changing, which is quite often. With a "height function", this recalculation will be avoided.

**Comment by Akryum** [comment]
I will soon release a mixin and a component that allows having unknown height items in RecycleList. Not sure if this will be necessary after that. 😉

**Comment by mikkiek** [comment]
@Akryum 

Hello, is "unknown height" feature implemented or can we expect to get such great feauture?
PS. i'm my case height really unknown it can be just 20px or 600px, depends on content(text) of block.

**Comment by Akryum** [comment]
@mikkiek Soon.

**Comment by xerosanyam** [comment]
we need this asap

**Comment by philippkuehn** [comment]
How much money do you need? 😅

**Comment by vinerz** [comment]
Hey @Akryum! Don't want to be that guy, but this feature would be truly amazing for us right now.
Could you just give us a hint about what to implement in order to achieve this effect?

**Comment by Akryum** [comment]
Closing as DynamicScroller can handle anything you throw at it with unknown heights!
